src/partition.ml

```diff
@@ -0,0 +1,94 @@
+open Encoding
+module H_contents = Irmin.Hash.Typed (Hash) (Contents)
+
```

Comment:
You are also missing some types to define the two kind of encoding for integers.

Reply:
Are we encoding integers in fixed-length in inodes?

Reply:
in the spec we are